### ISSUE TITLE
[DOC] Feedback thread on tutorial notebooks

### ISSUE THREAD

**Issue by fkiraly** [alan-turing-institute/sktime#1447]
This issue is for collecting feedback on the tutorial notebooks.

Any feedback is highly appreciated, positive or critical; can be high-level (e.g., too long where, helpful why, confusing how), or concrete, e.g., content that should be added, changed, removed, shortened, structured differently, etc.

**Comment by fkiraly** [comment]
from #1446, we ought to include sections on tags and estimator lookup in all tutorials;
we ought to include a section on multivariate forecasting in the forecasting tutorial

**Comment by fkiraly** [comment]
@gepitis, did you want to post some feedback on the tutorial notebooks? We've opened this just for you! (well, and also others who may like to provide such feedback, but you triggered the issue)

**Comment by shubhamkarande13** [comment]
Hi all!
I have gone through the first Forecasting notebook which is intended for people new to forecasting.
One suggestion for the notebooks; Is it possible to add a small definition or glossary section for easy reference of terms that a beginner like me would find difficult to understand?

For example: terms like _forecasting horizon_, _prediction intervals_ etc.

**Comment by fkiraly** [comment]
> For example: terms like forecasting horizon, prediction intervals etc.

Great idea, @shubhamkarande13!

There is already a glossary in the sktime docs, but it´s a bit incomplete
https://www.sktime.org/en/latest/glossary.html
and non-formal.

I was working on a sci reference, but got sidetracked by maintenance efforts.

**Comment by claudia-hm** [comment]
Hello!
I've just gone through the [Loading and working with data in sktime](https://www.sktime.org/en/stable/examples/loading_data.html) tutorial and I thought that I could provide some feedback. In overall, the tutorial is easy-to read and provides many examples of transforming time series datasets to different typical formats. Some comments:
* All examples have integer time-index, and very often time series come with datetime format. I would appreciate an explanation about what date formats .ts files accept and an example.
* I believe this tutorial would be a great place where to introduce the sktime built-in datasets, and the load_\<dataset\> methods, which I found very useful.

Regards!

**Comment by shubhamkarande13** [comment]
> > For example: terms like forecasting horizon, prediction intervals etc.
> 
> Great idea, @shubhamkarande13!
> 
> There is already a glossary in the sktime docs, but it´s a bit incomplete https://www.sktime.org/en/latest/glossary.html and non-formal.
> 
> I was working on a sci reference, but got sidetracked by maintenance efforts.

@fkiraly I would like to contribute by adding relevant terms to the glossary and learn in the process! Please let me know how we can move forward!

**Comment by fkiraly** [comment]
@claudia-hm, thanks for the feedback!

I just looked and I feel the tutorials are outdated, and there is a duplication with "user guide".
For instance, the "data formats" explained in the tutorial are old, see a newer notebook here:
https://github.com/alan-turing-institute/sktime/blob/main/examples/AA_datatypes_and_datasets.ipynb
Let me open an issue about this!

**Comment by fkiraly** [comment]
@claudia-hm, I've just reviewed the three different places where there are notebooks or user guides...
There are actually *tree places* 😱 how did this happen.

Anyway, I've suggested a plan how this could be improved, nice contribution opportunity: https://github.com/alan-turing-institute/sktime/issues/2127

**Comment by fkiraly** [comment]
@shubhamkarande13, thanks!

>@fkiraly I would like to contribute by adding relevant terms to the glossary and learn in the process! Please let me know how we can move forward!

May I suggest a very similar alternative? The tutorials and user guides are a bit of a mess, would you like to start moving them to a clean state, we could then increase the level of content over splicing/merging, up to writing the missing tutorials?

**Comment by h-t-w** [comment]
Hi. 

I am currently working through the forecasting tutorial. Quick observations:
-  Broken link under _Forecasting with sktime_ heading: https://www.sktime.org/en/stable/examples/01a_forecasting_sklearn.ipynb
- Formatting is off:

- Should probably be MD list

![image](https://user-images.githubusercontent.com/100365268/162437104-c28e8d38-babe-487c-b722-dc3058019682.png) 

- Should be link

![image](https://user-images.githubusercontent.com/100365268/162437308-52dbaf7d-3eb0-4305-bae2-dc96e68150db.png)

Best

**Comment by fkiraly** [comment]
@h-t-w, thanks for your feedback!

I've updated the forecaster tutorial here: https://github.com/alan-turing-institute/sktime/pull/2620

Feedback on the updated notebook would be appreciated.

**Comment by hilal-g** [comment]
Hello,

I read through the [Loading data into sktime](https://www.sktime.org/en/latest/examples/loading_data.html) tutorial, and I have a few [suggestions](http://localhost:8888/notebooks/loading_data_04.ipynb#Issue-1). 

1) Replace `A .ts file include two main parts: * header information * data` with `A .ts file includes two main parts: 1) header information and 2) data`. 
2) This one is a question regarding instances. What is the maximum number of dimensions?
3) The section "Representing data with .ts files" talks about the differences between a @timeStamps label of True and False. A possible update would be to add headings (i.e. @timeStamps=True and @timeStamps=False) with the relevant description for each condition below its respective heading.
4) As far as I understand, the word `obser` refers to `observation`. 

Best regards

**Comment by fkiraly** [comment]
Thanks, @hilal-g!

**Comment by CadePGCM** [comment]
Can you add a notebook that shows the functionality/use of MultiplexTransformer, OptionalPassThrough, TransformerPipeline, FeatureUnion, YtoX, etc.

Something exploring the space of transformation unions / compositions would be extremely helpful. This also might make sense as general functions to add to the library (union all transformations in the library), or some kind of (greedy/genetic?) algorithm to span the combinations space.

**Comment by fkiraly** [comment]
> Can you add a notebook that shows the functionality/use of MultiplexTransformer, OptionalPassThrough, TransformerPipeline, FeatureUnion, YtoX, etc.

Yes! I was working on it:
https://github.com/alan-turing-institute/sktime/pull/1705
but there is always something more urgent - bugs, releases, docstrings, admin, etc... :-(

Help is appreciated (search for "good first issues" 😄 ) - anything that takes pressure off the various other places makes it more likely that we write nice notebooks.

> Something exploring the space of transformation unions / compositions would be extremely helpful.

You can already use things like the `ForecastingGridSearch` and the `MultiplexForecaster` to do AutoML in the space, see forecasting tutorial.

**Comment by fkiraly** [comment]
@CadePGCM, good news for you!

We are writing a transformers & pipelines tutorial together with @miraep8 and the pywatts team (@benHeid, @kalebphipps, @smeisen) in preparation for a potential presentation at pydata.
Notebook and video recording be ready by Christmas 😄

**Comment by janjbat** [comment]
Hi,
I am currently working forecasting task. I am interested  to understand,  how one to tune hyperparameters and do backtesting at the same time, I might have missed it but could not find anything similar in the docs.  To be more specific, I want to do:

**Comment by janjbat** [comment]
Hello,

I am currently working on a forecasting task where I am interested  to run tuning of hyperparameters and backtesting simultaneously.  I might have overlooked it, but could not find anything similar in the docs or tutorials, so far.  
The main idea is to do a weekly hyperparamter tuning (taking past 12 weeks data and spiting it into training  (11 weeks) and test/validation (1week) sets)  and use the tuned model to produce daily forecasts for the whole next week by refitting the already tuned model each day. At the end of the next week repeat the process again.  It would be much appreciated if you could suggest or provide examples how backtesting can be implemented with sktime for such use cases?

Thank you and best regards,

B

**Comment by 3lle4** [comment]
Hello!
I'm doing some of your tutorials and I found a small bug in [02c_classification_multivariate_inceptiontime](https://github.com/sktime/sktime/blob/main/examples/02c_classification_multivariate_inceptiontime.ipynb):
In the current version, the last cell does not work. It seems that the parameter name `nb_filters` of the `InceptionTimeClassifier()` has been changed to `n_filters`.

Best

**Comment by fkiraly** [comment]
> I'm doing some of your tutorials and I found a small bug

Thanks for reporting. It's odd that this did not get caught by the tests, afaik they run all notebook cells. We'll look into this.

**Comment by jmwhyte** [comment]
With so many functions available for time-series classification, it would be good to have some more guidance, say in the introductory text at the top of help pages. Here's one good example of a useful tip, from the page for `ComposableTimeSeriesForestClassifier`:
"Parameters:

    estimator: Pipeline 
    A pipeline consisting of series-to-tabular transformations and a decision tree classifier as final estimator. 
"
We know that there is a section on "series-to-tabular transformations", so we might realise that this is the place to start reading. 

Compare this with something like the page for `TimeSeriersForestClassifier`:
"A time series forest is an ensemble of decision trees built on random intervals. Overview: Input n series length m."

If we have panel data, we can maybe work out that we need to combine our multivariate data for each instance into a univariate series. So, we can use `ColumnConcatenator`. But, is there anything else we can do? How would we work out the valid choices of transformers to pipeline together? As we try to extend our knowledge from sklearn methods to sktime, do we need to be careful to use a sktime pipeline, rather than a sklearn one?

I'm sure that once people gain more experience, this is easier to work out. But at the start, with so many different approaches (and other issues around reproducibility, see other comments), there's quite a knowledge gap to bridge just getting started. 

I suspect it might be possible to give other guidelines on which conditions are suitable for certain choices of transformer/classifier? Maybe some choices are better depending on whether you have thousands of data points or just hundreds (and find that some transformers produce NaN that stop the fitting process)?

**Comment by fkiraly** [comment]
@jmwhyte, are you thinking of sth like this

https://scikit-learn.org/stable/tutorial/machine_learning_map/index.html

just for sktime?

**Comment by fkiraly** [comment]
FYI @marrov - similar to what you have suggested as well

**Comment by jmwhyte** [comment]
Something like that sklearn diagram would be a good start, but I am sure there is also benefit in more detail at the function level.

**Comment by madhuri723** [comment]
Hi, 
When I was learning from the tutorial, I found ForecastingHorizon a bit confusing, especially the 'is_relative' part. Other forecasting tools usually need just one thing, but this one was different. It took me a bit to figure out that 'is_relative' is there to make things easier. I suggest making the explanation of 'is_relative' clearer so that it's easier for everyone to understand.

**Comment by madhuri723** [comment]
@fkiraly I can take up this work.

**Comment by fkiraly** [comment]
@madhuri723, thanks! Please go ahead!

**Comment by Prakruthi12345** [comment]
Hi, I have gone over the introductory notebook (00_sktime_intro), and it was very informative! I have come up with a couple of suggestions for the notebook and have included them in this document: https://docs.google.com/document/d/1w4lI7m5kWupSZCipr7N7HRPGbNimKAWCnzcdTvnVni4/edit 

Thanks, and please let me know what you think!

**Comment by fkiraly** [comment]
@Prakruthi12345, nice!

Could you please move these suggestions to a GitHub issue?

**Comment by Prakruthi12345** [comment]
@fkiraly will do, thanks!

**Comment by iamSathishR** [comment]
The code snippet (of 2.2.4 Time Series Classification - simple evaluation vignette) seems that the import statement for KNeighborsTimeSeriesClassifier (from sktime.classification.distance_based import KNeighborsTimeSeriesClassifier) is duplicated, appearing both in line 1 & line 9. Suggesting to remove the redundant (line 9) for any possible confusion

**Comment by fkiraly** [comment]
thanks! Would you like to make a PR?
Otherwise we can do it later too.

**Comment by iamSathishR** [comment]
> thanks! Would you like to make a PR? Otherwise we can do it later too.

Sure!! I would do a PR

**Comment by Syn-Eon** [comment]
Tutorial Feedback - Forecasting Basics
Tutorial : Getting Started with Forecasting (NaiveForecaster)

Hi,
I recently worked through the basic forecasting tutorial using the airline dataset.
it had clear progression and was a good choice for a dataset (easy to understand passenger trends).
The visualization helped me understand what forecasting does.

With that being said, however i think i had issues with the fact that in 
the tutorial the `fh` parameter wasn't explained before being used
also it was not clear why `fh=range(1,13)` vs `fh=12.
I didn't quite understand what "strategy='last'" means at first

Suggestions 
1. Add 1-2 sentence explanation of `fh` before first usage
2. Show what happens with different strategies (last vs mean)
alternatively, maybe 
-Add more inline comments in the example code
- Maybe include a diagram of train/test split if possible

**Comment by aparajitaks** [comment]
Hi, I worked through the forecasting tutorial as a beginner and also cross-checked concepts using external resources.

Key gaps I observed:

1. Time index assumptions not clearly explained
   The tutorial assumes a proper time index but does not explicitly explain the importance of monotonic time indexing for forecasting.

2. Limited evaluation strategy
   The tutorial mainly uses a simple train-test split. It would be helpful to introduce backtesting / rolling validation, which is a more robust approach.

3. Evaluation metrics lack context
   Metrics are used but not well explained in terms of when and why to use them (e.g., MAPE vs RMSE depending on use case).

4. Exogenous variables not clearly addressed
   There is no clear explanation of how models handle exogenous features, or that some models may ignore them.

5. Transformations and inverse steps missing clarity
   If preprocessing steps are applied (e.g., transformations), the tutorial should clearly explain how to reverse them to interpret results correctly.

Suggestions for improvement:

* Add a short section explaining time index requirements
* Introduce backtesting with a simple example
* Include guidance on choosing evaluation metrics
* Add a note on model limitations (exogenous support)
* Provide a clear example of data transformation + inverse transformation


Overall, the tutorial is helpful, but adding these would make it much more beginner-friendly and aligned with real-world forecasting practices.
